gapis/api/vulkan/framegraph.go

```diff
@@ -30,29 +30,59 @@ import (
 	"github.com/google/gapid/gapis/service/path"
 )
 
-func newFramegraphBuffer(buf *BufferObjectʳ) *api.FramegraphBuffer {
+// backedByCoherentMemory returns true if the device memory object is backed by
+// memory where the VK_MEMORY_PROPERTY_HOST_COHERENT_BIT is set.
+// Note that there exists a generated subIsMemoryCoherent(), but it needs all
+// the arguments of generated functions, which are not easily retrievable here.
+func backedByCoherentMemory(state *State, mem DeviceMemoryObjectʳ) bool {
+	phyDevHandle := state.Devices().Get(mem.Device()).PhysicalDevice()
+	phyDevObj := state.PhysicalDevices().Get(phyDevHandle)
+	propFlags := phyDevObj.MemoryProperties().MemoryTypes().Get(int(mem.MemoryTypeIndex())).PropertyFlags()
```

Comment:
Only a little bit worried on the closely connected queries here, about whether there would be nil pointer error on edge cases. But if the auto-generated files actually handle the nil pointer well, or quite some tests have been done, I think it's fine though.

Reply:
Good point. I believe here we can assume that we process valid Vulkan, so any VkDeviceMemory will have a valid physical device, where we can lookup memory properties. The generated `subIsMemoryCoherent()` does not check for nil pointers either.